mailman_public_inbox/__init__.py

```diff
@@ -44,27 +47,43 @@ def __init__(self):
         # Read our specific configuration file
         archiver_config = external_configuration(
             config.archiver.public_inbox.configuration)
+        self.base_url = archiver_config.get('general', 'base_url')
         self.public_inbox_config = archiver_config.get('general', 'pi_config')
         self.public_inbox_home = archiver_config.get('general', 'pi_home')
         self.public_inbox_path = archiver_config.get('general', 'pi_path')
+        self.auto_create = archiver_config.get('general', 'pi_auto_create')
+        self.reload_command = archiver_config.get('general', 'pi_reload_command')
 
         self.pi_config = {}
 
+        if self.auto_create:
+            classhandler.handler(ListCreatedEvent, self.list_created_handler)
+            classhandler.handler(ListDeletingEvent, self.list_deleting_handler)
+
+    def _run_command(self, args, **kwargs):
+        env = os.environ.copy()
+        env['PI_CONFIG'] = self.public_inbox_config
+        env['HOME'] = self.public_inbox_home
+        env['PATH'] = self.public_inbox_path
+
+        if env in kwargs:
```

Comment:
env should be "env" here I assume. This just throws an error about env not being hashable, cause it itself is a dict.